work14/index.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+const modalBtn = document.getElementById("js_modalButton");
+const modal = document.getElementById("js_modal");
+const requestBtn = document.getElementById("js_requestButton");
+
+const url = "https://mocki.io/v1/d6da0b8a-3546-419e-aaec-05f3a247c6d0";
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/eu2f";
+
+// 下記は、myjson繋がらない時の固定値
+// const url = {
+//   "data": [
+//     {
+//       "a": "bookmark",
+//       "img": "img/1.png",
+//       "alt": "画像１",
+//       "text": "ブックマーク"
+//     },
+//     {
+//       "a": "message",
+//       "img": "img/2.png",
+//       "alt": "画像２",
+//       "text": "メッセージ"
+//     }
+//   ]
+// }
+
+async function init() {
+  renderLoading();
+  const data = await getListData();
+  renderListElement(data);
+}
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      throw new Error(`Server request failed:${response.statusText}`);
+    }
+    // 下記は、固定値をそのままpromiseの返り値とする
+    // const json = url
+    // return json
+  } catch (e) {
+    console.error(e)
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  let listData;
+  const wrap = document.getElementById("js_wrap");
+
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    removeLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  return listData;
+}
+
+function removeLoading() {
+  document.getElementById("js_loading").remove();
+}
+
+function renderListElement({ data }) {
```

Comment:
Q.
Why is the argument `{data}`?
It seems to me that it would be better to simply use `data`, but if you have any reason for this, I would like to hear it.:bow:

Reply:
As you may know, we are using split assignment.
By setting the acquired variable "data" to "{data}", the object "data" can be called without describing "data.data", so split assignment was used.

Is it the answer?

[分割代入](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

[JavaScriptでObjectのKeyに変数展開する](https://note.com/vixer93/n/n1d3896fb0bd9)

Reply:
Was it a split assignment!
I have heard this before, but I didn't really understand it, so I will study it again.
Thank you for teaching me :raised_hands:

Reply:
Thank you for checking this.
I saw Akira-san's code.

> return result.data;

I felt that the code was easier to read when using.
```
async function fetchData() {
    try {
        const url = 'https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=ambjpfxjl00e50wa639kwndq1ofq3iuykdrv98ge';
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error('サーバーエラーです');
        }
        const result = await response.json();
        return result.data;
    } catch (responseError) {
        console.error(responseError);
        displayMessage(responseError);
    }
}
```